packages/indexer/src/sinks/drizzle/drizzle.test.ts

```diff
@@ -0,0 +1,207 @@
+import {
+  type MockBlock,
+  MockClient,
+  type MockFilter,
+} from "@apibara/protocol/testing";
+import { eq, sql } from "drizzle-orm";
+import { drizzle } from "drizzle-orm/node-postgres";
+import { serial, text } from "drizzle-orm/pg-core";
+import { Client } from "pg";
+import { beforeAll, beforeEach, describe, expect, it } from "vitest";
+import { type Int8Range, drizzle as drizzleSink, pgTable } from ".";
+import { useSink } from "../../hooks";
+import { run } from "../../indexer";
+import { generateMockMessages } from "../../testing";
+import { getMockIndexer } from "../../testing/indexer";
+
+const test_table = pgTable("test_table", {
+  id: serial("id").primaryKey(),
+  data: text("data"),
+});
+
+const client = new Client({
+  connectionString: "postgres://postgres:postgres@localhost:5432/postgres",
+});
+
+await client.connect();
+
+const db = drizzle(client);
+
+describe("Drizzle Test", () => {
+  beforeAll(async () => {
+    // drop test_table if exists
+    await db.execute(sql`DROP TABLE IF EXISTS test_table`);
+    // create test_table with db
+    await db.execute(
+      sql`CREATE TABLE test_table (id SERIAL PRIMARY KEY, data TEXT, _cursor INT8RANGE)`,
+    );
+  });
+
+  beforeEach(async () => {
+    await db.delete(test_table).execute();
+  });
+
+  it("should insert data", async () => {
+    const client = new MockClient<MockFilter, MockBlock>((request, options) => {
+      return generateMockMessages(5);
+    });
+
+    const sink = drizzleSink({ database: db });
+
+    const indexer = getMockIndexer({
+      sink,
+      override: {
+        transform: async ({ context, endCursor, block: { data } }) => {
+          const { transaction } = useSink({ context });
+
+          const db = transaction(endCursor);
```

Comment:
```suggestion
          const { db } = useSink({ context });
```

This won't be needed after you make the other changes.